src/components/Nav/index.tsx

```diff
@@ -60,7 +63,12 @@ const Nav = () => {
             ms={{ base: 3, xl: 8 }}
           >
             {/* avoid rendering desktop Menu version on mobile */}
-            {isDesktop && <Menu hideBelow="md" sections={linkSections} />}
+
+            {isClient && isDesktopFlag ? (
```

Comment:
I think just adding `isClient` might have done the trick, since it would be `false && _` and the rest wouldn't even be evaluated. Only when on the client.

Then again, I think it's probably best to have an appropriate `isDesktop` that checks for `window` itself as you've added